### PR TITLE
Ataymano/python interface fix

### PR DESCRIPTION
**Pull request by ataymano**

### CODE REVIEW

File: reinforcement_learning/doc/python/python.dox [hunk 1]
```diff
@@ -58,28 +58,37 @@ Note: `https://aka.ms/rl_client_python` points to a binary targeting x64 Linux P
...
 \param context_json Contains action, action features and context features in JSON format
+\param deferred Identifies if action is deferred or not. Deferred one requires additional call of report_action_taken to be invovled in training.
 \exception RuntimeError Thrown if an internal error occurs
 \return model_id, chosen_action_id, actions_probabilities
 \return model_id - Get the model id. Every call to choose action is associated with a unique model used to predict. A unique model id is associated with each unique model. (This is set internally by the API)
 \return chosen_action_id - ID of action chosen by model
 \return actions_probabilities - List of pairs (action_id, probability) ranked by the model's prediction
 
-\fn rl_client::live_model::choose_rank(context_json)
+\fn rl_client::live_model::choose_rank(context_json, deffered = False)
```
Comment (yannstad) [inline reinforcement_learning/doc/python/python.dox]: typo: `deffered` => `deferred`

File: reinforcement_learning/doc/python/python.dox [hunk 2]
```diff
@@ -58,28 +58,37 @@ Note: `https://aka.ms/rl_client_python` points to a binary targeting x64 Linux P
...
 \memberof rl_client::live_model
 \brief Choose an action, given a list of actions, action features and context features. The inference library chooses an action by creating a probability distribution over the actions and then sampling from it.
 \param context_json Contains action, action features and context features in JSON format
+\param deferred Identifies if action is deferred or not. Deferred one requires additional call of report_action_taken to be invovled in training.
```
Comment (yannstad) [inline reinforcement_learning/doc/python/python.dox]: typo: `invovled` => `involved`

File: reinforcement_learning/examples/python/basic_usage.py [hunk 3]
```diff
@@ -16,23 +16,31 @@ def main():
     model = rl_client.live_model(config, test_cb)
     model.init()
 
-    event_id = "event_id"
+    uuid = "event_id"
```
Comment (yannstad) [inline reinforcement_learning/examples/python/basic_usage.py]: why did you change the var name?

File: reinforcement_learning/examples/python/basic_usage.py [hunk 4]
```diff
@@ -16,23 +16,31 @@ def main():
...
 
-    model_id, chosen_action_id, actions_probabilities, event_id = model.choose_rank(context, deferred = True)
+    model_id, chosen_action_id, actions_probabilities, event_id = model.choose_rank(context, deferred = False)
+    model.report_action_taken(event_id)
+    print("event_id: " + event_id)
+    print("model_id: " + model_id)
+    print("chosen action id: " + str(chosen_action_id))
+    print("actions probabilities list: " + str(actions_probabilities))
+
+    model_id, chosen_action_id, actions_probabilities = model.choose_rank(context, deferred=True, event_id='asd')
```
Comment (yannstad) [inline reinforcement_learning/examples/python/basic_usage.py]: Maybe a more realistic event_id example is preferable?
Like for example: `e379df0b66fe48d1839e2a85a793c944y23fmJ`

File: reinforcement_learning/examples/python/basic_usage.py (hunk 3 above)
Comment (tyclintw) [reply reinforcement_learning/examples/python/basic_usage.py]: Yes, let's just leave the original naming for clarity.

File: reinforcement_learning/examples/python/basic_usage.py (hunk 4 above)
Comment (tyclintw) [inline reinforcement_learning/examples/python/basic_usage.py]: [nit] maintain the original argument ordering (context, event_id, deferred)

File: reinforcement_learning/doc/python/python.dox (hunk 1 above)
Comment (marco-rossi29) [inline reinforcement_learning/doc/python/python.dox]: deffered -> deferred

File: reinforcement_learning/examples/python/basic_usage.py (hunk 3 above)
Comment (marco-rossi29) [inline reinforcement_learning/examples/python/basic_usage.py]: For clarity, I would do:
```
import uuid  # add this on top of file

uuid = str(uuid.uuid4())
```

File: reinforcement_learning/examples/python/basic_usage.py (hunk 4 above)
Comment (marco-rossi29) [reply reinforcement_learning/examples/python/basic_usage.py]: I agree with Yann and Tyler. I would do:
import uuid
uuid = str(uuid.uuid4())
model_id, chosen_action_id, actions_probabilities = model.choose_rank(context, event_id=uuid, deferred=True)

Comment (marco-rossi29) [inline reinforcement_learning/examples/python/basic_usage.py]: I agree with Yann and Tyler. I would do:
import uuid
uuid = str(uuid.uuid4())
model_id, chosen_action_id, actions_probabilities = model.choose_rank(context, event_id=uuid, deferred=True)

Comment (yannstad) [reply reinforcement_learning/examples/python/basic_usage.py]: Actually, i thought we said that the deferred action logic will not be exposed except for advanced users.
Maybe we can omit to give an example where deferred is true?

Comment (marco-rossi29) [reply reinforcement_learning/examples/python/basic_usage.py]: That's a good point actually. @ataymano what do you think?

Comment (ataymano) [reply reinforcement_learning/examples/python/basic_usage.py]: are we exposing example? Even if we do, it is secondary thing compare to documentation in my opinion. Should I also remove it from docs?

Comment (ataymano) [reply reinforcement_learning/examples/python/basic_usage.py]: it was intentionally, to test that order is not important for named variables

Comment (ataymano) [reply reinforcement_learning/examples/python/basic_usage.py]: I can reorder though

Comment (yannstad) [reply reinforcement_learning/examples/python/basic_usage.py]: Neither one nor the other IMO
